cassandraprotocol/frame/decoder.go

```diff
@@ -45,27 +56,29 @@ func (r RawHeader) String() string {
 // one must either call DecodeBody or DiscardBody to fully read or discard the body contents.
 func (c *Codec) DecodeHeader(source io.Reader) (*RawHeader, error) {
 	if versionAndDirection, err := primitives.ReadByte(source); err != nil {
-		return nil, fmt.Errorf("cannot decode header version and direction: %w", err)
+		return nil, err
 	} else {
 		isResponse := (versionAndDirection & 0b1000_0000) > 0
 		version := versionAndDirection & 0b0111_1111
 		header := &RawHeader{
 			IsResponse: isResponse,
 			Version:    version,
 		}
+		var streamId uint16
 		if version < cassandraprotocol.ProtocolVersionMin || version > cassandraprotocol.ProtocolVersionMax {
 			return nil, fmt.Errorf("unsupported protocol version: %v", version)
 		} else if header.Flags, err = primitives.ReadByte(source); err != nil {
-			return nil, fmt.Errorf("cannot decode header flags: %w", err)
+			return nil, err
```

Comment:
Same here, I prefer the old messages as they are more explicit about what went wrong.

Reply:
I also think it is clearer if we enrich the message as Alex is suggesting.

Reply:
See the thread here https://github.com/datastax/go-cassandra-native-protocol/pull/5#discussion_r512689303